### ISSUE TITLE
Ability to run raw ftp commands directly

### ISSUE THREAD

**Issue by stancarney** [secsy/goftp#8]
It isn't uncommon for some industries utilizing FTP to make use of SITE commands (i.e. banking). Goftp does not allow the ability to run raw FTP commands, the required functions are not exported, so I wrapped pconn.sendCommand() in order to expose the ability to send arbitrary commands to the server.

https://github.com/stancarney/goftp/commit/7335f60db808f171b77976d4701221c776f3c37c

Is it something you might want to add to the library?

**Comment by muirmanders** [comment]
I'd prefer not to expose that since you could mess up the internal state of the connection depending what commands you run. It's also dangerous since each raw command could go to a different connection in the connection pool, leading to unexpected results.

Can you give some more details on your use case (like the particular command you need to run)? If we can't come up with something safe that solves the use case, I would rather directly expose the persistentConn as a separate lower-level connection you create independent of the Client. It would only be a single connection, and not interact with other connections at all.

**Comment by duke-4** [comment]
I need that functionality as well, in my private fork (that contains some more customization) i added a simple RawCommand function. My use cases at the moment are the commands:
* SITE VERS - to show the server software
* SITE SEARCH - search for directories

This commands may not be official supported in a standard, but some FTP server allow you to add custom commands that trigger scripts.

**Comment by stancarney** [comment]
My usage is also limited to SITE commands. 

i.e. 
* SITE standard=none
* SITE parm=blah

The underlying server technology is a product from OpenText (formerly known as GXS) and it is very common in the banking industry for file exchanges. Typical usage over FTPs has the client running several SITE commands before performing a transfer or a directory listing. The SITE commands are used to pass additional details about the files in question.

I guess a RunSiteCmd would also work if you are worried that somebody trying to run a PASV or other command would screw the connection up. To my knowledge SITE commands only have string inputs and outputs.

Maybe something like this?

```go
func (c *Client) SiteCmd(cmd string) (int, string, error) {
	pconn, err := c.getIdleConn()
	if err != nil {
		return 0, "", err
	}

	defer c.returnConn(pconn)

	return pconn.sendCommand(fmt.Sprintf("SITE %s", cmd))
}
```

An attempt to run PASV or another command would just result in a 500 being returned instead of messing up the connection.

**Comment by muirmanders** [comment]
Please look at #9. I added an "OpenRawConn" method that returns a separate RawConn you can use to run control or data commands. Let me know if you think that interface looks good.

**Comment by muirmanders** [comment]
Merged #9.

**Comment by stancarney** [comment]
That looks great! I will give it a test out later tonight. (sorry I thought I posted this message yesterday, but here it sits)
